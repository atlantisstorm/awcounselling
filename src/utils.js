export const initState = {
  first_name: "",
  surname: "",
  email: "",
  message: "",
  status: "",
  statusTextStyle: ""
};
  
export const statusMessages = {
  sent: "Your message has been sent.",
  sending: "Sending...",
  error: "Please ensure all fields are filled in and that your email address is correct!"
};

export const sendEmail = ({ formState, setFormState }) => {
    const firstName = formState.first_name;
    const surname = formState.surname;
    const email = formState.email;
    const message = formState.message;
    const emailRegex = /\S+@\S+\.\S+/;

    if (firstName.length > 0 && surname.length > 0 && emailRegex.test(email) > 0 && message.length) {
      const xhr = new XMLHttpRequest();
      xhr.open("POST", "/conform", true);
    
      xhr.setRequestHeader("Content-type", "application/json");
      xhr.onreadystatechange = function() {
        if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
          setFormState({
            ...formState, 
            status: statusMessages.sent,
            statusTextStyle: "success"
           });
        }
      };

      const data = {
        name: `${firstName} ${surname}`,
        email: email,
        message: message
      };

      xhr.send(JSON.stringify(data));
      setFormState({
        ...formState, 
        status: statusMessages.sending,
        statusTextStyle: "warning"
       });
    }
    else {
      setFormState({
        ...formState, 
        status: statusMessages.error,
        statusTextStyle: "danger"
       });      
    }    
}