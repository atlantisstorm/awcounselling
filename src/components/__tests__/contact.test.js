import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Contact from '../contact';
import { statusMessages }  from '../../utils';

describe('Contact component', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Contact />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display error if submit with empty fields.', () => {
    const { container, getByTestId } = render(
      <Contact />
    );

    expect(container.innerHTML).not.toContain(statusMessages.error);

    const submitButton = getByTestId("submit-button");
    fireEvent.click(submitButton);

    expect(container.innerHTML).toContain(statusMessages.error);
  });  
});