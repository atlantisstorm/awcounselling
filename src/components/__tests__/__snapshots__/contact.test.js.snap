// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component should match snapshot 1`] = `
<DocumentFragment>
  <h5>
    Contact
  </h5>
  <p>
    To contact me, please either fill in the contact form, use my email address, or call me by telephone.  My practise hours are 9am - 5pm (Monâ€“ Thurs), but I am also available during evenings and weekends to help ensure my services fit around your needs. I am also available to facilitate online counselling sessions via Skype or Zoom.
  </p>
  <div
    class="row"
  >
    <div
      class="col-lg-6"
    >
      <form
        action=""
        method=""
      >
        <div
          class="form-row mb-3"
        >
          <div
            class="col"
          >
            <input
              class="form-control"
              data-testid="form-firstname"
              name="first-name"
              placeholder="First name"
              type="text"
            />
          </div>
          <div
            class="col"
          >
            <input
              class="form-control"
              data-testid="form-surame"
              name="surname"
              placeholder="Surname"
              type="text"
            />
          </div>
        </div>
        <div
          class="form-row mb-3"
        >
          <div
            class="col"
          >
            <input
              class="form-control"
              data-testid="form-email"
              name="email"
              placeholder="email"
              type="text"
            />
          </div>
        </div>
        <div
          class="form-row mb-3"
        >
          <div
            class="col"
          >
            <textarea
              class="form-control"
              data-testid="form-comment"
              name="comment"
              placeholder="Place your comment/question here"
              rows="5"
            />
          </div>
        </div>
        <div
          class="form-row mb-3"
        >
          <div
            class="col"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      class="col-lg-6"
    >
      <h5>
        Aine Wilson Counselling
      </h5>
      <p>
        14 Park Hill
        <br />
        Tattinderry
        <br />
        Maguiresbridge
        <br />
        Co. Fermanagh
        <br />
        BT94 4AD
      </p>
      <p>
        <span
          class="font-weight-bold"
        >
          t.
        </span>
         028 6772 1046
        <br />
        <span
          class="font-weight-bold"
        >
          m.
        </span>
         07792 304 731
        <br />
        <span
          class="font-weight-bold"
        >
          m.
        </span>
         00353 (0) 863 086 381
        <br />
        <span
          class="font-weight-bold"
        >
          e.
        </span>
         ainewilson12@hotmail.com
      </p>
    </div>
  </div>
</DocumentFragment>
`;
